### ISSUE TITLE
Unable to use pairwise ranking with qid format

### ISSUE THREAD

**Issue by linkin8834** [dmlc/xgboost#4096]
Hi all,

I tried to train xgboost models locally using the pairwise loss function. I have a file that using the qid format as shown in https://xgboost.readthedocs.io/en/latest/tutorials/input_format.html#query-id-columns which is reproduced below:
```
1 qid:1000 101:1.2 102:0.03
0 qid:1000 1:2.1 10001:300 10002:400
0 qid:1200 0:1.3 1:0.3
1 qid:1200 0:0.01 1:0.3
0 qid:3000 0:0.2 1:0.3
1 qid:3000 3:-0.1 10:-0.3
0 qid:3000 6:0.2 10:0.15
```

Then, I use the following to train the model:
```
src/xgboost/xgboost pairwise.conf
```
but I am getting the following error:
```
[00:16:17] rabit/include/rabit/./internal/../../dmlc/./logging.h:300: 
[00:16:17] src/data/./row_block.h:177: Check failed: 
offset.back() == value.size() || value.size() == 0
```

The pairwise.conf file is shown below:
```
booster = gbtree
objective=rank:pairwise
eval_metric = ndcg
eval_metric = logloss
eval_metric = map
num_round = 10
save_period = 10
data = "train.txt"
test:data = "test.txt"
```

I have confirmed that
* The query ID (qid) strictly increases monotonically
* Each row comes with a qid

I am wondering if there is something that I missed? 

Thanks!

**Comment by hcho3** [comment]
@linkin8834 Most likely it is the version issue. Make sure that you are using XGBoost 0.80 or later.

**Comment by hcho3** [comment]
@linkin8834 Can you run `make clean` and try again? If that fails, try doing a clean checkout (`git clone --recursive https://github.com/dmlc/xgboost -b v0.81`).

**Comment by linkin8834** [comment]
@hcho3 Thanks for your suggestions. Yes `make clean` works for me. Let me try to do the pairwise ranking again. Thanks!
